pytorch_lightning/trainer/trainer.py

```diff
@@ -730,12 +730,16 @@ def _call_and_handle_interrupt(self, trainer_fn: Callable, *args: Any, **kwargs:
                 self.state.status = TrainerStatus.INTERRUPTED
                 self._call_callback_hooks("on_keyboard_interrupt")
                 self._call_callback_hooks("on_exception", exception)
+                for logger in self.loggers:
+                    logger.finalize("failed")
```

Comment:
Great. I think we should follow up on this PR with a small refactor that defines an enum for statuses.